### PR TITLE
Tab for uploader code generation

### PR DESCRIPTION
**Pull request by ErikApption**
- minor rework in file explorer to add a container level to the u/d options
- new uploader code generation in new tab
- configuration parameter to hide the new uploader tab

### CODE REVIEW

File: Portal/src/Datahub.Portal/Controllers/PrivateStorageController.cs [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+        _logger = logger;
+        _pubFileService = pubFileService;
+        this.dataRetrievalService = dataRetrievalService;
+    }
+
+    public IActionResult HelloWorld()
+    {
+        _logger.LogDebug("Unauthenticated hello world");
+        return Ok("hello world");
+    }
```
Comment (yjmrobert) [inline Portal/src/Datahub.Portal/Controllers/PrivateStorageController.cs]: 🌎

File: Portal/src/Datahub.Portal/Controllers/PrivateStorageController.cs [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+        {
+            folderpath = "nrcan-rncan.gc.ca/alexander.khavich",
+            filename = "serious.gif"
+        };
+
+        _logger.LogDebug($"Downloading {filemd.filename}");
+
+        var uri = await dataRetrievalService.DownloadFile(DataRetrievalService.DEFAULT_CONTAINER_NAME, filemd,null);
+        return Redirect(uri.ToString());
+    }
```
Comment (yjmrobert) [inline Portal/src/Datahub.Portal/Controllers/PrivateStorageController.cs]: Just highlighting the tests are still here 😄

**Review by yjmrobert — CHANGES_REQUESTED** [review]
Looks good, just need to restructure the folders to match the other projects

```
Datahub/
├─ Portal/
│  ├─ src/
│  ├─ test/
├─ Uploader/
│  ├─ src/
│  │  ├─ Uploader.Shared/
│  │  ├─ Uploader.MAUI/
```
Something like the above

File: Portal/src/Datahub.Portal/Controllers/PrivateStorageController.cs (hunk 1 above)
Comment (yjmrobert) [reply Portal/src/Datahub.Portal/Controllers/PrivateStorageController.cs]: Is this a health check?

File: Portal/src/Datahub.Portal/Controllers/PrivateStorageController.cs (hunk 2 above)
Comment (ErikApption) [reply Portal/src/Datahub.Portal/Controllers/PrivateStorageController.cs]: that's a merge error - not sure why there are there

File: Portal/src/Datahub.Portal/Controllers/PrivateStorageController.cs (hunk 1 above)
Comment (ErikApption) [reply Portal/src/Datahub.Portal/Controllers/PrivateStorageController.cs]: another merge error

**Comment by yjmrobert** [comment]
> Looks good, just need to restructure the folders to match the other projects
> 
> ```
> Datahub/
> ├─ Portal/
> │  ├─ src/
> │  ├─ test/
> ├─ Uploader/
> │  ├─ src/
> │  │  ├─ Uploader.Shared/
> │  │  ├─ Uploader.MAUI/
> ```
> 
> Something like the above

Suggestion to rename from `Uploader` to `Desktop` 👍

**Comment by ErikApption** [comment]
@yjmrobert I renamed MAUI to Desktop

**Pull request merged**